src/main/proto/ess.proto

```diff
@@ -0,0 +1,64 @@
+syntax = "proto3";
+option java_multiple_files = true;
+package org.ess;
+
+service LightningStorageService {
+  rpc get(GetRequest) returns (GetResponse);
+  rpc put(PutRequest) returns (PutResponse);
+  //TODO: Add getKeysSummary (Gives version numbers for all keys in a store) and list by prefix API's
+}
+
+message GetRequest {
+  /*
+    All API's operate within a single storeId.
+    StoreId is keyspace identifier. Ref: https://en.wikipedia.org/wiki/Keyspace_(distributed_data_store)
+    It is up to clients to use single or multiple store's for their usage.
+   */
+  string storeId = 1;
+  string key = 2;
+}
+message GetResponse {
+  bool success = 1;
+  KeyValue value = 2;
```

Comment:
is it useful to return just one value, or does it make more sense to return everything with a certain key prefix?  for VLS at least we always want to get a complete snapshot of the storage for a node when we do perform a `get`.  does LDK intend to make single item retrievals?

Reply:
regarding "return everything with a certain key prefix" => i thought about it but this makes for an unbounded query on server-side hence we should be avoiding it
single get does mean some additional work for client.

Regarding "list by key prefix" => this in intended to be solved by listKeys api which will be a paginated api. For LDK as well, we have use-case of reloading parts of snapshot. (This will be done after computing version difference of keys b/w client and server-side)

I had added "Batch" operation support as "future item" in doc, but that also won't solve this problem because there will still be some batch limit. Client needs to use versioning to ensure consistent state instead of relying on getting everything at once in consistent fashion.
Another reason, I was being bit conservative about batch operations is because we intend to support storing values upto ~5MB, even a 10 item get might mean 50MB response depending on client-application.

Let me know if that works.

Side note: Currently this api can be used with any of the approach mentioned in doc, that is "Independent State Updates", "Global State Counter", "Incremental State Updates". LDK will be using "Global State Counter"

Reply:
I don't see "Independent State Updates" in the doc.  However, that sounds like the approach that VLS would use.  But would be interested in hearing more.

So I think the plan with `listKeys` would work, but we will need this sooner rather than later, since otherwise there is no way to discover what is in the store.

Reply:
Yes, LDK needs listKeys and getKeysSummary immediately as well.
Just didn't include it in this commit. Wanted close on basic get and put first. 

Heading was named slightly different. Added "Independent State Update" to that heading.